src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -0,0 +1,655 @@
+                this.SingleOption(CodeStyleOptions.OtherOperationParentheses, new CodeStyleOption<OtherOperationParenthesesPreference>(OtherOperationParenthesesPreference.RemoveIfUnnecessary, NotificationOption.Suggestion)));
+
+        private async Task TestAsync(string initial, string expected, bool whenClarifyingPrecedence)
+        {
+            await TestInRegularAndScriptAsync(initial, expected, options: AllRemove);
+
+            if (whenClarifyingPrecedence)
+            {
+                await TestInRegularAndScriptAsync(initial, expected, options: AllRequireForClarity);
+            }
+            else
+            {
+                await TestMissingAsync(initial, parameters: new TestParameters(options: AllRequireForClarity));
+            }
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestVariableInitializer_Always()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = [||](1);
+    }
+}", new TestParameters(options: AllIgnore));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestArithmeticRequiredForClarity1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = 1 + [||](2 * 3);
+    }
+}", new TestParameters(options: ArithmeticRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestArithmeticRequiredForClarity2()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        int x = a || [||](b && c);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = a || b && c;
+    }
+}", parameters: new TestParameters(options: ArithmeticRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestLogicalRequiredForClarity1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        int x = a || [||](b && c);
+    }
+}", new TestParameters(options: LogicalRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestLogicalRequiredForClarity2()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        int x = a + [||](b * c);
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        int x = a + b * c;
+    }
+}", parameters: new TestParameters(options: LogicalRequireForClarity));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestArithmeticNotRequiredForClarityWhenPrecedenceStaysTheSame1()
```

Comment:
As discussed elsewhere, the precedence doesn't change but the order of evaluation does, which can be dangerous in several cases.